### ISSUE TITLE
etcd 3.5.9 still uses gobin instead of go run

### ISSUE THREAD

**Issue by kfirufk** [etcd-io/etcd#16001]
### What happened?

gobin is old and archived and the project should fully.. not support it anymore and use go run instead :)

there was an old pull request that was merged https://github.com/etcd-io/etcd/pull/13673
but it still uses go bin.

### What did you expect to happen?

i expected that build.sh will not require gobin anymore

### How can we reproduce it (as minimally and precisely as possible)?

well if i empty GOPATH and i run build.sh it creates github.com/myitcv/gobin directory.

you can see that the test files are still looking for gobin at scripts/test_lib.sh

### Anything else we need to know?

in general i was trying to update an ebuild for gentoo for etcd 3.5.9, there is a way to create a dependency archive for the build using GOMODCACHE="${PWD}"/go-mod go mod download -modcacherw and archiving it and using it in the build, but in this case it's not enough because the build.sh script fails on missing gobin. and gentoo doesn't allow connecting remotely to download anything else as security measure.

### Etcd version (please run commands below)

etcd v3.5.9

### Relevant log output

```shell
>>> Emerging (1 of 1) dev-db/etcd-3.5.9::tuxin
 * etcd-3.5.9.tar.gz BLAKE2B SHA512 size ;-) ...                                                          [ ok ]
 * etcd-3.5.9-deps.tar.xz BLAKE2B SHA512 size ;-) ...                                                     [ ok ]
>>> Unpacking source...
>>> Unpacking etcd-3.5.9.tar.gz to /var/tmp/portage/dev-db/etcd-3.5.9/work
>>> Unpacking etcd-3.5.9-deps.tar.xz to /var/tmp/portage/dev-db/etcd-3.5.9/work
go mod verify
all modules verified
>>> Source unpacked in /var/tmp/portage/dev-db/etcd-3.5.9/work
>>> Preparing source in /var/tmp/portage/dev-db/etcd-3.5.9/work/etcd-3.5.9 ...
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/dev-db/etcd-3.5.9/work/etcd-3.5.9 ...
>>> Source configured.
>>> Compiling source in /var/tmp/portage/dev-db/etcd-3.5.9/work/etcd-3.5.9 ...
% 'env' 'GO111MODULE=off' 'go' 'get' 'github.com/myitcv/gobin'
stderr: github.com/myitcv/gobin (download)
stderr: created GOPATH=/var/tmp/portage/dev-db/etcd-3.5.9/homedir/go; see 'go help gopath'
stderr: cd /var/tmp/portage/dev-db/etcd-3.5.9/homedir/go/src/github.com/myitcv
stderr: git clone -- https://github.com/myitcv/gobin /var/tmp/portage/dev-db/etcd-3.5.9/homedir/go/src/github.com/myitcv/gobin
stderr: # cd /var/tmp/portage/dev-db/etcd-3.5.9/homedir/go/src/github.com/myitcv; git clone -- https://github.com/myitcv/gobin /var/tmp/portage/dev-db/etcd-3.5.9/homedir/go/src/github.com/myitcv/gobin
stderr: Cloning into '/var/tmp/portage/dev-db/etcd-3.5.9/homedir/go/src/github.com/myitcv/gobin'...
stderr: fatal: unable to access 'https://github.com/myitcv/gobin/': Could not resolve host: github.com
stderr: package github.com/myitcv/gobin: exit status 128
```

**Comment by kkkkun** [comment]
https://github.com/etcd-io/etcd/pull/13673 is solved in main.  In release-3.5 it's still use gobin.

**Comment by jmhbnz** [comment]
Hey @ahrtr - Do you think it's appropriate for #13673 to be backported?

**Comment by ahrtr** [comment]
> Hey @ahrtr - Do you think it's appropriate for #13673 to be backported?

I am supportive on it.

**Comment by kkkkun** [comment]
assign it to me. I would take a look.

**Comment by jmhbnz** [comment]
Linked backport pr has been merged. @kfirufk this fix will be released in the next patch release of etcd `3.5` which will be `3.5.10`. Alternatively you can check out either the `main` or `release-3.5` branches and this issue will no longer be present.

Will close this now, thanks for working on this @kkkkun 🙏🏻
